src/commands/index.js

```diff
@@ -0,0 +1,427 @@
+        toolbar.push({ attributes: { class: "fa fa fa-th-large", title: 'Unmerge cells' }, command: "table-unmerge-cells" })
+      }
+      return toolbar;
+    }
+
+    const getTableToolbar = (component) => {
+      const tb = component.get('toolbar');
+      let settingExists = tb.find(o=> o.command === 'open-table-settings-modal');
+      if(!settingExists) {
+        tb.push(
+          {
+              command: 'open-table-settings-modal',
+              attributes: {class: 'fa fa-cog', title: 'Settings'},
+          }
+        );
+      }
+      return tb;
+    }
+  
+    editor.on('component:selected', component => {
+      if (component.get('type') == 'cell' || component.get('type') == 'th') {
+        component.set('toolbar', getCellToolbar()); // set a toolbars
+      }
+
+      if(component.get('type') == 'customTable'){
+        component.set('toolbar', getTableToolbar(component));
+      }
+    });
+  
+    cmd.add('table-show-columns-operations', () => {
+      updateTableToolbarSubmenu('columns', 'rows');
+    });
+  
+    cmd.add('table-show-rows-operations', () => {
+      updateTableToolbarSubmenu('rows', 'columns');
+    });
+  
+    cmd.add('table-select', editor => {
+      editor.runCommand('core:component-exit');
+      editor.runCommand('core:component-exit');
+    });
+  
+    cmd.add('table-insert-row-above', editor => {
+      let selected = editor.getSelected();
+      if (selected.is('cell')) {
+        let rowComponent = selected.parent();
+        let table = selected.parent().parent();
+        rowComponent.parent().components().add({
+          type: 'row',
+          components: [...Array(rowComponent.components().length).keys()].map(() => ({ type: 'cell' }))
+        }, {
+          at: rowComponent.collection.indexOf(rowComponent)
+        });
+        let tableEl = table.getEl();
+        let rowCount = tableEl.getAttribute('data-nRows');
+        tableEl.setAttribute('data-nRows', rowCount + 1);
+        editor.selectRemove(selected);
+        setTimeout(function() {
+ editor.select(selected); 
+}, 50);
+      }
+    });
+  
+    cmd.add('table-insert-row-below', editor => {
+      let selected = editor.getSelected();
+      if (selected.is('cell') || selected.is('th')) {
+        let rowComponent = selected.parent();
+        let table = selected.parent().parent();
+        rowComponent.parent().components().add({
+          type: 'row',
+          components: [...Array(rowComponent.components().length).keys()].map(() => ({ type: 'cell' }))
+        }, {
+          at: rowComponent.collection.indexOf(rowComponent) + 1
+        });
+        let tableEl = table.getEl();
+        let rowCount = tableEl.getAttribute('data-nRows');
+        tableEl.setAttribute('data-nRows', rowCount + 1);
+        editor.selectRemove(selected);
+        setTimeout(function() {
+ editor.select(selected); 
+}, 50);
```

Comment:
Seems like something broke the formatting. Seems to be issue with all `setTimeout`

Reply:
jslint did that, I can manually correct it, not familiar enough with it to get the rules to not do that.